vm/process/systemVM.go

```diff
@@ -114,15 +116,8 @@ func (s *systemVM) RunSmartContractCall(input *vmcommon.ContractCallInput) (*vmc
 		}, nil
 	}
 
-	lockedGas, err := s.handleAsyncStepGas(input)
-	if err != nil {
-		return &vmcommon.VMOutput{
-			ReturnCode:    vmcommon.UserError,
-			ReturnMessage: err.Error(),
-			GasRemaining:  lockedGas,
-		}, nil
-	}
-
+	lockedGas := input.GasLocked
+	input.GasProvided -= lockedGas
```

Comment:
The locked gas should have already been deducted. `input.GasProvided` is the complete gas available to `RunSmartContractCall()`, without deducting `lockedGas`. This can be deleted.

Reply:
moved this out from systemvm and arwen vm as well.